### PR TITLE
Use cds-icon instead of clr-icon for icon component

### PR DESCRIPTION
**Pull request by GuessWhoSamFoo**
**What this PR does / why we need it**:
`clr-icon` has reached deprecation in clarity 4 and will receive no future updates for 5+.

This PR changes the icon component to use `cds-icon`. It adds helper methods with descriptions for various custom behaviors. Small storybook inconsistencies are addressed.

It also removes a lot of custom logic to generate correct styling based on class attributes. This should be part of 0.19 as many new icons are now available.

**Which issue(s) this PR fixes**
- Partially addresses #2213 

**Special notes for your reviewer**:

Internal usage of `clr-icon` within other components should be updated as well, but this does not need to be tied to a release.

Signed-off-by: Sam Foo <foos@vmware.com>

### CODE REVIEW

File: web/src/stories/icon.stories.mdx [hunk 1]
```diff
@@ -20,7 +20,7 @@ c.Config.Badge = component.BadgeDanger
 <Meta title="Components/Icon" component={IconComponent} />
 
 <Canvas withToolbar>
```
Comment (mklanjsek) [inline web/src/stories/icon.stories.mdx:22]: Not sure the last `<p>` above is accurate any more, can't we remove Clarity 4 reference (same for signpost)?

File: web/src/stories/icon.stories.mdx [hunk 2]
```diff
@@ -29,14 +29,13 @@ c.Config.Badge = component.BadgeDanger
...
           config: {
             shape: "user",
-            size: "65",
+            size: "md",
             direction: "up",
             flip: "horizontal",
             solid: true,
             status: "danger",
             inverse: false,
             badge: "danger",
```
Comment (mklanjsek) [inline web/src/stories/icon.stories.mdx:38]: Can you add badge color and label, we always try to include all properties (same for signpost).

**Review by mklanjsek — CHANGES_REQUESTED** [review]
Looks good, just a few small tweaks. Can please create an issue to address switch for internal icons, would be good to have them in sync ASAP.

File: web/src/app/modules/shared/models/content.ts [hunk 3]
```diff
@@ -591,6 +591,8 @@ export interface IconView extends View {
     inverse: boolean;
     badge: string;
     color: string;
+    badgeColor: string;
+    label: string;
```
[the remark this answers is not in the document]
Comment (GuessWhoSamFoo) [reply web/src/app/modules/shared/models/content.ts:595]: Go returns empty strings, optional wouldn't be meaningful here

**Comment by GuessWhoSamFoo** [comment]
Fixed improper setting of aria-label and loading of `--color` and `--badge-color` styles. Removed references to clarity 4

**Review by mklanjsek — APPROVED** [review]
🚢

**Pull request merged**